app/models/alchemy/page.rb

```diff
@@ -380,11 +376,15 @@ def copy_children_to(new_parent)
     #
     def publish!
       current_time = Time.current
-      update_columns(
+      published_hash = {
         published_at: current_time,
         public_on: current_time,
         public_until: nil
-      )
+      }
+      unless first_published_at
```

Comment:
Since this method is only called if someone hits the "publish" button, we also need to take care for cases where someone publishes the page by setting the `public_on` date in the page settings overlay. 

So, I think we should put this in a method to avoid duplication.